### DISCUSSION TITLE
Use custom column for identifying in methods like `hasRole`

### DISCUSSION THREAD

**Discussion by Kazuto** [spatie/laravel-permission#1846] [Ideas]
Hey guys,

Many things can be configured in `config/permission.php`. It would be great to be able to choose the identifier for methods like `hasRole` as well so that we can use the standard `name` column or a custom column e.g `slug`.

I prefer having proper text in the name column like `Shareholder`, `Developer` or `Super Administrator` and have another unique column called `slug` for identification which would result in `shareholder`, `developer`, and `super-administrator`. 
Being only able to use the `name` column kinda restricts the casing of names as `role:Super Admininstrator` just looks wrong in code and `role:super-administrator` would fit much better, but then `super-administrator` would look wrong in UI.

**Comment by PaolaRuby** [comment]
You could make a PR for that, because surely no one else is going to do
